engine/class_modules/sc_death_knight.cpp

```diff
@@ -5924,6 +5946,17 @@ struct obliterate_strike_t : public death_knight_melee_attack_t
     }
   }
 
+  int n_targets() const override
+  {
+    // Unable to find the # of targets in spell data for death's due, but it does cause each obliterate hit to cleave. 2020-Nov-6
+    int num_targets = 0;
+    if ( p() -> covenant.deaths_due -> ok() )
+    {
+      num_targets = p() -> in_death_and_decay() ? 2 : 0;
+    }
+    return num_targets;
```

Comment:
Death's Due's tooltip references 315442 effect#2 for obliterate's target count increase. 
Moreover, I also noticed that the obliterate strike spells (222024 1H, 325461 2H, and 66198 OH) all have a chain_multiplier and chain_targets value of 1 in spelldata, so you should probably use that.

Reply:
Actually, it seems that Death and Decay contains data for Obliterate in its effect#4 
```#4 (id=871411)   : Apply Aura (6) | Add Flat Modifier (107): Spell Targets (17)
                   Base Value: 0 | Scaled Value: 0 | PvP Coefficient: 1.00000 | Misc Value: 17 | Target: Self (1)
                   Affected Spells: Obliterate (49020), Obliterate Off-Hand (66198), Obliterate (222024), Obliterate (325461)
                   Family Flags: 49```
This value of 0 is increased to 1 by 315442, making the chain_targets value 2

Reply:
Makes sense how it's all being done on blizz's end.  Is there a better way that we can accomplish this in simc than just adding 315442#2 to 325461#1.chain_target()?  It ends up skipping the interm 49020#4 (obliterate).

Reply:
Well the spelldata driven way would be to return the damaging effect's chain_target() + the value from 49020#4 if inside dnd + the value from 315442#2 if inside dnd and death's due is talented, while making sure that if the result is 1, it returns 0 (to save computation time, "true" ST abilities have aoe=0 that let them skip target lists, target specific state, etc. ).

That said, your inital ``num_targets = p() -> in_death_and_decay() ? 2 : 0;`` works very well too and I doubt that it's something Blizzard will change in the near future. If you're afraid of that you could also compute the total value during construction and return either that or 0 on runtime based on death and decay/covenant.